pkg/datapath/linux/bandwidth/bandwidth.go

```diff
@@ -69,11 +67,23 @@ func (m *manager) defines() (defines.Map, error) {
 	return cDefinesMap, nil
 }
 
-func (m *manager) DeleteEndpointBandwidthLimit(epID uint16) error {
+func (m *manager) UpdateBandwidthLimit(epID uint16, bytesPerSecond uint64) {
```

Comment:
I think I would prefer not having wrappers like this, but rather use the database rather directly. This makes it more obvious to reader where the data eventually lands and also hints that it can be inspected with "cilium-dbg statedb". The less layers we have the better.

But for now I don't feel like wiring DB and this table into the endpoint package...

Reply:
Personally, I have a bit of an opposing perspective. It is possible to have too many layers, but the wrappers are helpful for the following reasons:

1. They act as an abstraction to improve readability. When reading through code, it's easier to understand what `UpdateBandwidthLimit` does than `EdtTable.Insert(txn, bwmap.NewEdt(epId, bytesPerSecond))`. The former is more intuitive and aligns with Effective Go practices.
2. They act as an implementation buffer. If the interface of statedb has a breaking change, or if we discover a bug that needs to be fixed, it's much easier to update a single wrapper than the usage of the database across every consumer of the bandwidth manager.
3. As a developer consuming the bandwidth manager, I shouldn't have to understand how it works underneath the hood. It doesn't matter to me where the data eventually lands when all I need to do is adjust the bandwidth. If it's important to understand how the bandwidth adjustment works, then I can read the wrapper's implementation.
4. Having a single wrapper method with the correct implementation is much easier to consume than needing to write out a database transaction. As a developer, I'd feel more comfortable using the wrapper than the database, because I'd be less worried about making a mistake.

> also hints that it can be inspected with "cilium-dbg statedb".

IMO, this should be the responsibility of the documentation and not the responsibility of the code.